test/views/pages/map/dynamic_map_test.dart

```diff
@@ -0,0 +1,286 @@
+    await tester.tap(
+      find.descendant(
+        of: bottomBar,
+        matching: find
+            .byType(NavigationDestination)
+            .at(NavigationbarRoutes.map.index),
+      ),
+    );
+    await tester.pumpAndSettle();
+
+    final element = tester.element(find.byType(HomePage));
+    return ProviderScope.containerOf(element);
+  }
+
+  /// Verifies that the [pins] and [posts] match.
+  /// This means that they have the same id, and the same location. The order
+  /// of the pins and posts does not matter.
+  void expectPinsAndPostsMatch(
+    List<MapPinDetails> pins,
+    List<PostFirestore> posts,
+  ) {
+    // Compare the ids
+    final pinsId = pins.map((pin) => pin.id.value);
+    final postIds = posts.map((post) => post.id.value);
+    expect(pinsId, unorderedEquals(postIds));
+
+    // Compare the positions
+    final pinsPosition = pins.map((pin) {
+      final pos = pin.position;
+      return GeoPoint(pos.latitude, pos.longitude);
+    });
+    final postPositions = posts.map((post) => post.location.geoPoint);
+    expect(pinsPosition, unorderedEquals(postPositions));
+  }
+
+  group("Option selection", () {
+    late List<PostFirestore> nearbyPosts;
+    late List<PostFirestore> userPosts;
+
+    late Map<MapSelectionOptions, List<PostFirestore>> expectedPostsForOption;
+
+    setUp(() async {
+      await setUserFirestore(fakeFirestore, testingUserFirestore);
+
+      nearbyPosts = postGenerator.generatePostsAtDifferentLocations(
+        GeoPointGenerator.generatePositions(userPosition0, 10, 0),
+      );
+      await setPostsFirestore(nearbyPosts, fakeFirestore);
+
+      final farPosts = postGenerator.generatePostsAtDifferentLocations(
+        GeoPointGenerator.generatePositions(userPosition0, 0, 10),
+      );
+      await setPostsFirestore(farPosts, fakeFirestore);
+
+      userPosts = postGenerator.createUserPosts(
+        testingUserFirestoreId,
+        userPosition1,
+        10,
+      );
+      await setPostsFirestore(userPosts, fakeFirestore);
+
+      expectedPostsForOption = {
+        MapSelectionOptions.nearby: nearbyPosts,
+        MapSelectionOptions.myPosts: userPosts,
+      };
+    });
+
+    testWidgets("Correct default option", (tester) async {
+      final container = await beginTest(tester);
+
+      final currentOption = container.read(
+        mapSelectionOptionsViewModelProvider,
+      );
+      expect(
+        currentOption,
+        equals(MapSelectionOptionsViewModel.defaultMapOption),
+      );
+    });
+
+    /// Verifies that the [option] is indeed the current option of the view-model
+    /// (in the view of the [container]), and that the pins returned by the view-model
+    /// are indeed the one expected in [expectedPostsForOption]. If [expectedPostForOption]
+    /// does not contain the [option], then it is assumed that the expected posts
+    /// are an empty list.
+    Future<void> testOption(
```

Comment:
I think we could use a more explicit name for this function. Maybe something like: `testMapSelectionOptionAndPins`

Reply:
That's perfect ! Thank you ! 👍🏼